add-file/add-file.spec.yaml

```
@@ -0,0 +1,233 @@
+name: add-file-at-result-path
+description: Add file to all path with a search result. In this example, we add an Apache license beside each README.md file.
+
+# Find all repositories that contain a README.md file.
+on:
+  - repositoriesMatchingQuery: file:README.md count:200
+
+# Besides each match, add a readme file. Note that repository.search_result_paths is a helper variable provided by the spec templating language.
+steps:
+  - run: |
+        export ROOTDIR=`pwd`
+        for result in ${{repository.search_result_paths}};
+        do
+            export DIR=${result%/*};
+            cd $DIR
+            cat /tmp/apache.txt > LICENSE.md
+            cd $ROOTDIR
+        done
+
```

Maybe using pushd and popd it'd look a little simpler? (Warn: untested)
```suggestion
        for result in ${{repository.search_result_paths}};
        do
            pushd ${result%/*}
            cat /tmp/apache.txt > LICENSE.md
            popd
        done

```